### PR TITLE
Remove body from POST request in webhook

### LINKED ISSUE #5066: webhook lookup invalid json in body
I have a minio server which uses webhooks for notification.
When I'm starting the server, it sends a request to the webhook where the body is not a JSON however the content-type is specified as application/json.

## Expected Behavior
Body would respect the Content-type header. If Content-type says application/json then the body would be JSON also.

## Current Behavior
The following request is sent to my configured webhook when the minio server is starting:

```
POST /webhook HTTP/1.1
Host: 10.30.0.64:5000
User-Agent: Minio/DEVELOPMENT.2017-10-11T17-19-39Z (linux; amd64)
Content-Length: 32
Content-Type: application/json
Accept-Encoding: gzip
Connection: close

DEVELOPMENT.2017-10-11T17-19-39Z
```

Here, the body is not a valid JSON but the Content-Type says that it is a JSON, which is contradiction.

## Steps to Reproduce (for bugs)
1. use minio/minio:edge, but also works with minio/minio docker image
2. configure webhook as described in the documentation 
3. start the webhook daemon (here, I used `netcat -l -p 5000` for debugging purposes)
4. start minio server

### PR DESCRIPTION
**Pull request by nitisht**
## Description
If webhook notification is configured, Minio server tries to lookup the
webhook endpoint by making a POST request with body set as releasetag.
We can remove the body from the POST request as the POST body does not
add any specific value.

This discussion on IETF group says empty POSTs are okay
http://lists.w3.org/Archives/Public/ietf-http-wg/2010JulSep/0272.html

## Motivation and Context
Fixes: https://github.com/minio/minio/issues/5066

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by vadmeste** [comment]
I don't think webhook is working in master.

Besides, I was wondering why we are not sending clients notification data as we do with other notifications hooks.

**Comment by harshavardhana** [comment]
> I don't think webhook is working in master.
> 

Webhook is working fine. 

> Besides, I was wondering why we are not sending clients notification data as we do with other notifications hooks.

Not sure what does this mean..

**Comment by vadmeste** [comment]
> Webhook is working fine.

yes sorry.. one typo in mc alias, which was considered as local directory



> Not sure what does this mean..

I mean why are we sending empty data? we can send notification information instead as we do in NATs, Redis, MySQL etc.. (this is out of scope of this PR)

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
> I mean why are we sending empty data? we can send notification information instead as we do in NATs, Redis, MySQL etc.. (this is out of scope of this PR)

No we are sending empty to validate the endpoint @vadmeste is correctly configured.

File: cmd/notify-webhook.go [hunk 1]
```diff
@@ -107,8 +107,8 @@ func lookupEndpoint(urlStr string) error {
 		},
 	}
 
-	// Set content-type.
-	req.Header.Set("Content-Type", "application/json")
+	// Set content-length to zero as there is no payload.
+	req.Header.Set("Content-Length", "0")
```
Comment (harshavardhana) [inline cmd/notify-webhook.go]: we don't need to set like this explicitly. just set it as `req.ContentLength = 0`

File: cmd/notify-webhook.go [hunk 2]
```diff
@@ -91,10 +91,10 @@ func isNetErrorIgnored(err error) bool {
 }
 
 // Lookup endpoint address by successfully POSTting
-// a JSON which would send out minio release.
+// empty body.
 func lookupEndpoint(urlStr string) error {
-	minioRelease := bytes.NewReader([]byte(ReleaseTag))
-	req, err := http.NewRequest("POST", urlStr, minioRelease)
+	emptyBody := bytes.NewReader([]byte(""))
```
Comment (harshavardhana) [inline cmd/notify-webhook.go]: No need to have a new variable for this just do this 

```
http.NewRequest("POST", urlStr, bytes.NewReader([]byte("")))
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/notify-webhook.go (hunk 2 above)
Comment (nitisht) [reply cmd/notify-webhook.go]: done

File: cmd/notify-webhook.go (hunk 1 above)
Comment (nitisht) [reply cmd/notify-webhook.go]: done

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**